ocamltest/filecompare.ml

```diff
@@ -227,10 +227,20 @@ let check_file ?(tool = default_comparison_tool) files =
     else Unexpected_output
   end
 
+(* Best-effort hack to enable colored diff output *)
+let colordiff =
+  List.map @@ fun flag ->
+  let prefix = "--color=" in
+  if String.starts_with ~prefix flag then
+    let color = if Misc.Color.is_enabled () then "always" else "never" in
+    prefix ^ color
```

Comment:
This code is hard to understand without also reading at your configure.ac changes. Can you explain here in a comment was is going on?

(I think that the clean way would be to export a separate variable `diff_supports_color` variable, and when it is true add an extra `--color=[always|never]` parameter to diff_flags. I'm not convinced that your approach is better, it saves a bit of configure changes but makes things weirder down stream.)

Reply:
I've added this:

```
(* The configure script tests if the diffing tool supports the [--color=auto]
   flag (a no-op), and if so, passes it to [Ocamltest_config.diff_flags]. Find
   this flag, and replace its value with the color preference of the user. *)
```

Maybe it's too much of a hack, as you suggest.